src/core/pointcloud/qgspointcloudextentrenderer.cpp

```diff
@@ -134,6 +136,14 @@ void QgsPointCloudExtentRenderer::setFillSymbol( QgsFillSymbol *symbol )
 {
   mFillSymbol.reset( symbol );
 }
+void QgsPointCloudExtentRenderer::renderLabels( const QRectF &extent, const QString &text, QgsPointCloudRenderContext &context )
+{
+  const double textWidth = QgsTextRenderer::textWidth( context.renderContext(), labelTextFormat() ? *labelTextFormat() : QgsTextFormat(), QStringList() << text );
+  if ( textWidth < extent.width() )
+  {
+    QgsTextRenderer::drawText( extent, 0, Qgis::TextHorizontalAlignment::Center, QStringList() << text, context.renderContext(), labelTextFormat() ? *labelTextFormat() : QgsTextFormat(), true, Qgis::TextVerticalAlignment::VerticalCenter );
+  }
```

Comment:
This is a little inefficient, because QgsTextRender::textWidth requires parsing the text and determining document metrics and then this will be repeated in ::drawText.

Try the approach used here https://github.com/qgis/QGIS/blob/master/src/app/decorations/qgsdecorationgrid.cpp#L305-L308 and https://github.com/qgis/QGIS/blob/master/src/app/decorations/qgsdecorationgrid.cpp#L374 instead. That will parse the document and calculate metrics once upfront, and then reuse in the drawing.

It's considerably faster this way if you're drawing a lot of text

Reply:
Okay I didn't know about this limitation, the current version implements it.